### PR TITLE
Drop exemplars in distributor for tenants where exemplars are disabled

### PR DESCRIPTION
**Pull request by hi-rustin**
#### What this PR does
- Drop exemplars in distributor for tenants where exemplars are disabled
- Add test for it
- Fix other tests

#### Which issue(s) this PR fixes or relates to

Fixes https://github.com/grafana/mimir/issues/697

#### Checklist

- [x] Tests updated
- [x] `CHANGELOG.md` updated - the order of entries should be `[CHANGE]`, `[FEATURE]`, `[ENHANCEMENT]`, `[BUGFIX]`

### CODE REVIEW

File: pkg/distributor/distributor_test.go [hunk 1]
```diff
@@ -1576,12 +1608,16 @@ func TestDistributor_ExemplarValidation(t *testing.T) {
...
 	for testName, tc := range tests {
 		t.Run(testName, func(t *testing.T) {
+			limits := &validation.Limits{}
+			flagext.DefaultValues(limits)
+			tc.prepareConfig(limits)
+			ds, _, _ := prepare(t, prepConfig{
+				limits:          limits,
+				numDistributors: 1,
+			})
 			err := ds[0].validateSeries(now, tc.req.Timeseries[0], "user", false, tc.minExemplarTS)
```
[the remark this answers is not in the document]
Comment (56quarters) [reply pkg/distributor/distributor_test.go:1621]: I think you're right. Maybe the tests each used to only have a single timeseries, but some of them definitely have multiple now. Makes sense to update it to validate each series.

File: pkg/distributor/distributor.go [hunk 2]
```diff
@@ -569,26 +569,29 @@ func (d *Distributor) validateSeries(nowt time.Time, ts mimirpb.PreallocTimeseri
...
+					ts.Exemplars[i] = ts.Exemplars[last]
+				}
+				ts.Exemplars = ts.Exemplars[:last]
+				continue
+			}
+			i++
 		}
-		i++
+	} else {
+		ts.Exemplars = make([]mimirpb.Exemplar, 0)
```
Comment (56quarters) [inline pkg/distributor/distributor.go:594]: nit: I believe this can be replaced with
```suggestion
		ts.Exemplars = nil
```

**Review by 56quarters — APPROVED** [review]
LGTM, thanks for doing this! Just one small change besides the test issue you noted

**Review by replay — APPROVED** [review]
LGTM, thanks!

**Pull request merged**